internal/provider/resource_flink_artifact.go

```diff
@@ -31,15 +31,23 @@ func artifactResource() *schema.Resource {
 				Type:         schema.TypeString,
 				Required:     true,
 				ForceNew:     true,
-				Description:  "The display name of Flink Artifact",
+				Description:  "The unique name of the Flink Artifact per cloud, region, environment scope.",
 				ValidateFunc: validation.StringLenBetween(1, 60),
 			},
 			paramClass: {
 				Type:         schema.TypeString,
-				Required:     true,
+				Optional:     true,
 				ForceNew:     true,
 				Description:  "Java class or alias for the Flink Artifact as provided by developer.",
 				ValidateFunc: validation.StringMatch(regexp.MustCompile(pattern), "The class must be in the required format"),
+				Deprecated:   "No longer required.",
```

Comment:
We'll need to update this error message, see the example here:
* https://github.com/confluentinc/terraform-provider-confluent/commit/5d91a31931799fa9af1198a8ad7056d1ac844c7d#diff-3891edf4bd794ecf121bf45d76a5c56b74993731d044c21fbdb887f22319c709

Reply:
Not sure we wanted to update the Readme since we dont mention the class there anymore, but updated the resource accordingly

Reply:
Didn't we mention it in the previous version? See https://registry.terraform.io/providers/confluentinc/confluent/latest/docs/resources/confluent_flink_artifact